polkadot-host-spec/ag-runtimeentries.tm

```diff
@@ -470,16 +470,22 @@
 
   <subsection|<verbatim|BlockBuilder_inherent_extrinsics>>
 
-  Generate inherent extrinsics. The inherent data will vary from chain to
-  chain.
+  Generates the inherent extrinsics, which are explained in more detail in
+  section <reference|sect-inherents>. This function takes a SCALE encoded
+  hashtable as defined in section <reference|defn-scale-list> and returns an
+  array of extrinsics. The Polkadot Host must submit each of those to
+  <verbatim|BlockBuilder_apply_extrinsic>, described in section
+  <reference|sect-rte-apply-extrinsic>. This procedure is outlined in
+  algorithm <reference|algo-build-block>.
 
   \;
 
   <strong|Arguments>:
 
   <\itemize>
     <item>A <name|Inherents-Data> structure as defined in
-    <reference|defn-inherent-data>.
+    <reference|defn-inherent-data>. This function does not generate an
```

Comment:
I'm not sure if it is a good place for this comment. Maybe at the return value because it is about what the function is generating? Related to that is that, should you or not submit the babeslot inherent to the function or not. If not, then how babeslot is being recorded in header then? In any case it is good to explain how babeslot is getting added to the block.